### PR TITLE
Feature/508 notification with resource

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

- Changed a few model interfaces to base classes to avoid repeating properties. 
- Moved ignoreReservations and resourceId to base class => consequence: the properties will be included in all order endpoints including  `/status` and `/notifications`.  Do we want this, or should I move the properties to the relevant models? Have not made relevant repository changes yet. 
- New profile client to retrieve unit details
- Updated to contact point service to use both profile client and authorization service

## Related Issue(s)
- #508 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Notifications.Core/Integrations/IProfileClient.cs [hunk 1]
```diff
@@ -13,4 +13,12 @@ public interface IProfileClient
...
     /// <returns>A list of contact points for the provided national identity numbers </returns>
     public Task<List<UserContactPoints>> GetUserContactPoints(List<string> nationalIdentityNumbers);
+
+    /// <summary>
+    /// Retrieves the user registered contact points for a list of organization corresponding to a list of organization numbers
+    /// </summary>
+    /// <param name="organizationNumbers">The set or organizations to retrieve contact points for</param>
+    /// <param name="resourceId">The id of the resource to look up contact points for</param>
+    /// <returns>A list of organiation contact points containing user registered contact points</returns>
+    public Task<List<OrganizationContactPoints>> GetUserRegisteredOrganizationContactPoints(List<string> organizationNumbers, string resourceId);
```
Comment (acn-sbuad) [inline src/Altinn.Notifications.Core/Integrations/IProfileClient.cs:23]: A bit of a mouthful, and other suggestions as to what we can name it ?

Comment (SandGrainOne) [reply src/Altinn.Notifications.Core/Integrations/IProfileClient.cs:23]: 1. GetUserRegisteredContactPoints (just remove organization, it's implied based on input?)
2. GetRightHoldersContactPoints (right holders can also imply a user in association with another party)
3. GetYourContactInformationForOrganization (reuse the term used by the GUI where users edit the information)

Comment (acn-sbuad) [reply src/Altinn.Notifications.Core/Integrations/IProfileClient.cs]: Went for option 1 ✅

File: test/Altinn.Notifications.Tests/Notifications.Integrations/Profile/ProfileClientTests.cs [hunk 2]
```diff
@@ -82,7 +88,39 @@ public async Task GetUserContactPoints_FailureResponse_ExceptionIsThrown()
...
+
+    [Fact]
+    public async Task GetUserRegisteredContactPoints_FailureResponse_ExceptionIsThrown()
+    {
+        // Act
+        var exception = await Assert.ThrowsAsync<PlatformHttpException>(
+            async () => await _profileClient.GetUserRegisteredContactPoints(["12345678", "98754321"], "error-resource"));
+
+        Assert.StartsWith("ProfileClient.GetUserRegisteredContactPoints failed with status code", exception.Message);
+        Assert.Equal(HttpStatusCode.ServiceUnavailable, exception.Response?.StatusCode);
```
Comment (SandGrainOne) [inline test/Altinn.Notifications.Tests/Notifications.Integrations/Profile/ProfileClientTests.cs:120]: The Response property is not nullable. I'm a little picky here, but the question mark is basically adding an unnecessary if check. (Might be that the complier detects that it's unnecessary and exclude it for us.)

**Review by SandGrainOne — APPROVED** [review]
Approved with some nitpicking.

File: src/Altinn.Notifications/Models/NotificationOrderRequestBaseExt.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    /// Gets or sets the senders reference on the notification
+    /// </summary>
+    [JsonPropertyName("sendersReference")]
+    public string? SendersReference { get; set; }
+
+    /// <summary>
+    /// Gets or sets the list of recipients
+    /// </summary>
+    [JsonPropertyName("recipients")]
+    public List<RecipientExt> Recipients { get; set; } = new List<RecipientExt>();
```
[the remark this answers is not in the document]
Comment (acn-sbuad) [reply src/Altinn.Notifications/Models/NotificationOrderRequestBaseExt.cs:27]: Ran a quick test and as the property is non nullable, we are forced to set the default value `new List<RecipientExt>()` ensuring the property is never undefined, resulting in a bad request response never being returned to caller. 

That being said, our custom validation rules ensure that recipient is populated. We could add the required property to ensure the Open API spec is clear w.r.t which properties are required or not.

[the remark this answers is not in the document]
Comment (SandGrainOne) [reply src/Altinn.Notifications/Models/NotificationOrderRequestBaseExt.cs:27]: That's what I was thinking.

**Pull request merged**